src/clients/java/src/main/java/com/tigerbeetle/UInt128.java

```diff
@@ -172,5 +174,53 @@ public static byte[] asBytes(final BigInteger value) {
         return asBytes(bigintLsb.longValueExact(), bigintMsb.longValueExact());
     }
 
+    private static long ulidLastTimestamp = 0L;
+    private static final byte[] ulidLastRandom = new byte[80];
+    private static final SecureRandom ulidSecureRandom = new SecureRandom();
 
+    /**
+     * Generates a Universally Unique Lexicographically Sortable Identifier
+     * (<a href="https://github.com/ulid/spec">ULID</a>) as 16 bytes of a 128-bit value.
+     *
+     * The ULID returned always increases monotonically and is thread-safe.
+     *
+     * @return an array of 16 bytes representing the 128-bit value.
+     */
+    public static byte[] ULID() throws ArithmeticException {
```

Comment:
By convention unchecked exceptions should not be declared in the `throws` clause, but only documented.

```suggestion
    /**
     * Generates a Universally Unique Lexicographically Sortable Identifier
     * (<a href="https://github.com/ulid/spec">ULID</a>) as 16 bytes of a 128-bit value.
     *
     * The ULID returned always increases monotonically and is thread-safe.
     *
     * @throws ArithmeticException as per ULID specs, if the 80 bits of randomness overflows.
     * @return an array of 16 bytes representing the 128-bit value.
     */
    public static byte[] ULID() {
```